src/main/java/edu/kit/ipd/crowdcontrol/objectservice/database/operations/NotificationOperations.java

```diff
@@ -96,29 +111,38 @@ public Notification insertNotification(Notification toStore) throws IllegalArgum
                 Notification.NAME_FIELD_NUMBER,
                 Notification.DESCRIPTION_FIELD_NUMBER,
                 Notification.QUERY_FIELD_NUMBER,
-                Notification.CHECK_PERIOD_FIELD_NUMBER);
+                Notification.CHECK_PERIOD_FIELD_NUMBER,
+                Notification.SEND_ONCE_FIELD_NUMBER,
+                Notification.EMAILS_FIELD_NUMBER);
+
+        toStore.getEmailsList().forEach(email -> {
+            if (!EmailValidator.getInstance(false).isValid(email)) {
```

Comment:
We check if all required fields are present as well here, so I think it's the right place.